### PR TITLE
UserFunction(abc.ABC) -> UserFunction(abc.ABC, Callable)

### PR DESCRIPTION
**Pull request by ekalosak**
*Issue #, if available:*
NA

*Description of changes:*
On top of the context-in-OuterLoop::get_next_points() PR, this implements the Callable mixin for UserFunction. It means you can do `user_function(x)` in addition to `user_function.evaluate(x)`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by apaleyes** [comment]
Can you fix the commit history here, please? It includes changes from another PR

**Comment by ekalosak** [comment]
1. `git rebase -i e514476fdf5e39f322ae6a6c7730102e6865f529~`
2. Change bad commits to `drop` from `pick`
3. `git push --force`
^ in case anyone else needs the git recipe

File: emukit/core/loop/user_function.py [hunk 1]
```diff
@@ -21,11 +21,13 @@
 _log = logging.getLogger(__name__)
 
 
-class UserFunction(abc.ABC):
+class UserFunction(abc.ABC, Callable):
     """ The user supplied function is interrogated as part of the outer loop """
     @abc.abstractmethod
     def evaluate(self, X: np.ndarray) -> List[UserFunctionResult]:
         pass
+    def __call__(self, X: np.ndarray) -> List[UserFunctionResult]:
```
Comment (apaleyes) [inline emukit/core/loop/user_function.py:29]: A line break is missing here

**Comment by apaleyes** [comment]
I should have mentioned earlier, but can you please add a unit test to cover this?

**Comment by ekalosak** [comment]
Unrelated, but thoughts on adding box/flaky to pytest?

**Comment by apaleyes** [comment]
Emukit's test runs are normally consistent, so not yet. But yes, sometimes we do run into transient failures, so flaky might come handy. I don't know what's box.

**Comment by ekalosak** [comment]
Ah, box as in github.com/box/flaky. And, unit test ahoy.

**Comment by apaleyes** [comment]
Looks good to me! We will need either @MashaNaslidnyk or @mmahsereci to merge this though, so let's wait for them to chime in.

**Pull request merged**